htsget/htsget.go

```diff
@@ -0,0 +1,224 @@
+
+	"github.com/NBISweden/sda-cli/helpers"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help htsget` command
+var Usage = `
+USAGE: %s htsget [-dataset <datasetID>] [-filename <filename>] (-reference <referenceName>) [-htsgethost <htsget-hostname>] [-key <public-key-file>] (-outdir <dir>)
+
+htsget:
+    Htsget downloads files from the Sensitive Data Archive (SDA), using the
+	htsget server. A dataset and a filename must be provided in order to 
+	download the file. The files will be downloaded in the current
+    directory, if outdir is not defined.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help
+var ArgHelp = `
+    [dataset]
+        The ID of the dataset that the file is part of.
+    [filename]
+        The name of the file to download.
+	[reference]
+		The reference number of the file to download.
+	[htsgethost]
+		The hostname of the htsget server to use.
+	[key]
+		The public key file to use for the htsget request.`
+
+// Args is a flagset that needs to be exported so that it can be written to the
+// main program help
+var Args = flag.NewFlagSet("htsget", flag.ExitOnError)
+var datasetID = Args.String("dataset", "", "Dataset ID for the file to download")
+var fileName = Args.String("filename", "", "The name of the file to download")
+var referenceName = Args.String("reference", "", "The reference number of the file to download")
+var htsgetHost = Args.String("htsgethost", "", "The htsget host to use")
+var publicKeyFile = Args.String("key", "", "Public key file to use for htsget request")
+var configPath = Args.String("config", "",
+	"S3 config file to use for uploading.")
+var outDir = Args.String("outdir", "",
+	"Directory for downloading file.")
+
+type htsgetResponse struct {
+	Htsget struct {
+		Format string `json:"format"`
+		Urls   []struct {
+			URL     string `json:"url"`
+			Headers struct {
+				Range          string `json:"Range"`
+				UserAgent      string `json:"user-agent"`
+				Host           string `json:"host"`
+				AcceptEncoding string `json:"accept-encoding"`
+				Authorization  string `json:"authorization"`
+			} `json:"headers,omitempty"`
+		} `json:"urls"`
+	} `json:"htsget"`
+}
+
+// Htsget function downloads the files included in the urls_list.txt file.
+// The argument can be a local file or a url to an S3 folder
+func Htsget(args []string) error {
+
+	// Call ParseArgs to take care of all the flag parsing
+	err := helpers.ParseArgs(args, Args)
+	if err != nil {
+		return fmt.Errorf("failed parsing arguments, reason: %v", err)
+	}
+
+	var currentPath string
+	currentPath, err = os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current path, reason: %v", err)
+	}
+
+	if *datasetID == "" || *fileName == "" || *htsgetHost == "" || *publicKeyFile == "" {
+		return fmt.Errorf("missing required arguments, datasetID, fileName, htsgetHost and key are required")
```

Comment:
```suggestion
		return fmt.Errorf("missing required arguments, dataset, filename, htsgethost and key are required")
```
To be consistent with the flag names.